### ISSUE TITLE
Need to explicitely set $parser->list->idx = 0;

### ISSUE THREAD

**Issue by corbolais** [phpmyadmin/sql-parser#320]
Hello devs,

Thanks for this excellent validating parser.

Trying to iterate over the [list](https://develdocs.phpmyadmin.net/sql-parser/PhpMyAdmin/SqlParser/Parser.html) of [Tokens](https://develdocs.phpmyadmin.net/sql-parser/PhpMyAdmin/SqlParser/TokensList.html), it does not work as assumed: I first need to explicitely set 
```php
    $parser->list->idx = 0;
```
Only then the iteration is working (for me):
```php
    while ( NULL !== $token = $parser->list->getNextOfType(PhpMyAdmin\SqlParser\Token::TYPE_COMMENT) ) {
    [...]
    }
```
Bug or feature? Did I wrongly assume a different behaviour?

The docs don't specify anything in that regard (nor much on anything else). Any intention to improve on documentation besides "`Use the Source..`"?

Thank you for clarification.

**Comment by williamdes** [comment]
Hello @corbolais 

Thank you !

This is a good question, maybe @ibennetch or @devenbansod would have the answer ?

That said a code sample would help me search for you an answer

> The docs don't specify anything in that regard (nor much on anything else). Any intention to improve on documentation besides "Use the Source.."?

We are open for propositions to improve our documentation ;)

**Comment by ibennetch** [comment]
I unfortunately don't know the answer off the top of my head, but we will try to get an answer for you.

About the documentation, I agree that it needs to be improved. It hasn't been much of a priority, in part because we haven't realized the project has gained some usage outside of phpMyAdmin. This is something we can work towards improving.

**Comment by williamdes** [comment]
Hi @corbolais 
What where you trying to do ?
Can you provide a full example ?

**Comment by corbolais** [comment]
Hi @williamdes 

I remember trying to solve an issue that involved iterating over a list. The document stated the obvious, the rest I glimpsed from the source.

There was an error thrown however and to solve this it was required to initialize the index like so:

```
$parser->list->idx = 0;
```

I wanted to let you guys know this, just in case.

Of the top off my head I'm ignoring whether the requirement to init the index was due to some specific PHP version.

Cheers and greetings to Brittany, France!

A la prochaine..
